dialects/mysql/src/main/kotlin/app/cash/sqldelight/dialects/mysql/grammar/mixins/DropForeignKeyMixin.kt

```diff
@@ -0,0 +1,66 @@
+package app.cash.sqldelight.dialects.mysql.grammar.mixins
+
+import app.cash.sqldelight.dialects.mysql.grammar.psi.MySqlAlterTableDropForeignKey
+import com.alecstrong.sql.psi.core.SqlAnnotationHolder
+import com.alecstrong.sql.psi.core.psi.SqlColumnDef
+import com.alecstrong.sql.psi.core.psi.SqlColumnName
+import com.alecstrong.sql.psi.core.psi.SqlCompositeElementImpl
+import com.alecstrong.sql.psi.core.psi.SqlCreateTableStmt
+import com.alecstrong.sql.psi.core.psi.SqlTableConstraint
+import com.intellij.lang.ASTNode
+import com.intellij.psi.util.childrenOfType
+import com.intellij.psi.util.parentOfType
+
+abstract class DropForeignKeyMixin(
+  node: ASTNode,
+) : SqlCompositeElementImpl(node),
+  MySqlAlterTableDropForeignKey {
+  override fun annotate(annotationHolder: SqlAnnotationHolder) {
+    super.annotate(annotationHolder)
+    val foreignKey = childrenOfType<SqlColumnName>().single()
+    val foundColumnDef = foreignKey.getColumnDefOrNull()
+    if (foundColumnDef == null || !foundColumnDef.isForeignKey()) {
+      annotationHolder.createErrorAnnotation(
+        foreignKey,
+        "${foreignKey.name} is not a foreign key.",
+      )
+    }
+  }
+
+  private fun SqlColumnName.getColumnDefOrNull(): SqlColumnDef? {
```

Comment:
Should be moved to sql-psi to allow reusing getting the columnDef and its foreign status in other dialects too, will do it in another PR.